plugins/catalog-backend-module-bitbucket-server/src/providers/BitbucketServerEntityProvider.ts

```diff
@@ -204,19 +229,51 @@ export class BitbucketServerEntityProvider implements EntityProvider {
         if (this.config?.filters?.skipArchivedRepos && repository.archived) {
           continue;
         }
-        for await (const entity of this.parser({
+
+        const options: ParseOptions = {
           client,
           logger: this.logger,
           location: {
             type: 'url',
             target: `${repository.links.self[0].href}${this.config.catalogPath}`,
             presence: 'optional',
           },
-        })) {
-          result.push(entity);
-        }
+        };
+
+        await this.parseEntities(result, options, parseFunction);
       }
     }
     return result;
   }
+
+  private parse(options: ParseOptions) {
+    return this.parser(options);
+  }
+
+  private lastCalled: number | null = null;
+
+  private async parseEntities(
+    entities: Entity[],
+    options: ParseOptions,
+    parseFn: (opt: ParseOptions) => Promise<AsyncIterable<Entity>>,
+  ) {
+    const now = Date.now();
+    if (this.lastCalled !== null) {
+      const duration = now - this.lastCalled;
+      this.logger.debug(`Time since last call: ${duration}ms`);
```

Comment:
Hmm lots of debugging here, do we really need to keep it?

Reply:
removed